railties/test/application/middleware/exceptions_test.rb

```diff
@@ -171,5 +171,41 @@ def index
       assert_equal 400, last_response.status
       assert_match "Invalid query parameters", last_response.body
     end
+
+    test "displays statement invalid template correctly" do
+      controller :foo, <<-RUBY
+        class FooController < ActionController::Base
+          def index
+            raise ActiveRecord::StatementInvalid
+          end
+        end
+      RUBY
+      app.config.action_dispatch.show_exceptions = true
+      app.config.consider_all_requests_local = true
+
+      get "/foo"
+      assert_equal 500, last_response.status
+      assert_match "<title>Action Controller: Exception caught</title>", last_response.body
+      assert_match "ActiveRecord::StatementInvalid", last_response.body
+    end
+
+    test "displays statement invalid template correctly for plain text request" do
+      controller :foo, <<-RUBY
+        class FooController < ActionController::Base
+          def index
+            raise ActiveRecord::StatementInvalid
+          end
+        end
+      RUBY
+      app.config.action_dispatch.show_exceptions = true
+      app.config.consider_all_requests_local = true
+      app.config.action_dispatch.ignore_accept_header = false
+
+      get "/foo", {}, { "HTTP_ACCEPT" => "text/plain", "HTTP_X_REQUESTED_WITH" => "XMLHttpRequest" }
```

Comment:
Weird, but I was expecting this to work, but it did not:

`get "/foo", headers: { "HTTP_ACCEPT" => "text/plain", "HTTP_X_REQUESTED_WITH" => "XMLHttpRequest" }
`